test/common/shouldDo.test.js

```diff
@@ -0,0 +1,297 @@
+    });
+  });
+
+  context('Certain Days of the Week', () => {
+    it('leaves Daily inactive if day of the week does not match', () => {
+      dailyTask.repeat = {
+        su: false,
+        s: false,
+        f: false,
+        th: false,
+        w: false,
+        t: false,
+        m: false,
+      };
+
+      for (let weekday of [0, 1, 2, 3, 4, 5, 6]) {
+        day = moment().day(weekday).toDate();
+
+        expect(shouldDo(day, dailyTask, options)).to.equal(false);
+      }
+    });
+
+    it('leaves Daily inactive if day of the week does not match and active on the day it matches', () => {
+      dailyTask.repeat = {
+        su: false,
+        s: false,
+        f: false,
+        th: true,
+        w: false,
+        t: false,
+        m: false,
+      };
+
+      for (let weekday of [0, 1, 2, 3, 4, 5, 6]) {
+        day = moment().add(1, 'weeks').day(weekday).toDate();
+
+        if (weekday === 4) {
+          expect(shouldDo(day, dailyTask, options)).to.equal(true);
+        } else {
+          expect(shouldDo(day, dailyTask, options)).to.equal(false);
+        }
+      }
+    });
+
+    it('activates Daily on matching days of the week', () => {
+      expect(shouldDo(day, dailyTask, options)).to.equal(true);
+    });
+  });
+
+  context('Every X Weeks', () => {
+    it('leaves daily inactive if it has not been the specified number of weeks', () => {
+      dailyTask.everyX = 3;
+      let tomorrow = moment().add(1, 'day').toDate();
+
+      expect(shouldDo(tomorrow, dailyTask, options)).to.equal(false);
+    });
+
+    it('leaves daily inactive if on every (x) week on weekday it is incorrect weekday', () => {
+      dailyTask.repeat = {
+        su: false,
+        s: false,
+        f: false,
+        th: false,
+        w: false,
+        t: false,
+        m: false,
+      };
+
+      day = moment();
+      dailyTask.repeat[DAY_MAPPING[day.day()]] = true;
+      dailyTask.everyX = 3;
+      let threeWeeksFromTodayPlusOne = day.add(1, 'day').add(3, 'weeks').toDate();
+
+      expect(shouldDo(threeWeeksFromTodayPlusOne, dailyTask, options)).to.equal(false);
+    });
+
+    it('activates Daily on matching week', () => {
+      dailyTask.everyX = 3;
+      let threeWeeksFromToday = moment().add(3, 'weeks').toDate();
+
+      expect(shouldDo(threeWeeksFromToday, dailyTask, options)).to.equal(true);
+    });
+
+    it('activates Daily on every (x) week on weekday', () => {
+      dailyTask.repeat = {
+        su: false,
+        s: false,
+        f: false,
+        th: false,
+        w: false,
+        t: false,
+        m: false,
+      };
+
+      day = moment();
+      dailyTask.repeat[DAY_MAPPING[day.day()]] = true;
+      dailyTask.everyX = 3;
+      let threeWeeksFromToday = day.add(6, 'weeks').day(day.day()).toDate();
+
+      expect(shouldDo(threeWeeksFromToday, dailyTask, options)).to.equal(true);
+    });
+  });
+
+  context('Monthly - Every Day of the month', () => {
```

Comment:
Hmm. I fine with rewording, but every x day can be confused with every x days. This is representing every date (different than the day).